lib/config/ts-loader-config.ts

```diff
@@ -0,0 +1,69 @@
+import os from 'os'
+import webpack from 'webpack'
+import TsconfigPathsPlugin from 'tsconfig-paths-webpack-plugin'
+import ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin'
+import { CreateWebpackConfigOptions } from '../runtime/server'
+import { BuildTarget } from '../types'
+
+const disableCaching = String(process.env.BUILD_CACHE_DISABLED).toLowerCase() === 'true'
+const disableTypeCheck = process.env.START_FAST_MODE === 'true'
+
+const osCpus = os.cpus().length
+const threadLoaderCpus = !disableTypeCheck ? osCpus - 1 : osCpus
+
+const threadLoader = {
+    loader: 'thread-loader',
+    options: {
+        // there should be 1 cpu for the fork-ts-checker-webpack-plugin if its enabled
+        workers: threadLoaderCpus,
+    },
+}
+
+const cacheLoader = (cacheDirectory: string) => {
+    return {
+        loader: 'cache-loader',
+        options: {
+            cacheDirectory,
+        },
+    }
+}
+
+export function getTypeScriptWebpackRule(
+    plugins: webpack.Plugin[],
+    resolvePlugins: webpack.ResolvePlugin[],
+    options: CreateWebpackConfigOptions,
+    buildTarget: BuildTarget,
+): webpack.RuleSetRule {
+    if (!disableTypeCheck) {
+        plugins.push(new ForkTsCheckerWebpackPlugin({ checkSyntacticErrors: true })) // fork ts checking
+    }
+
+    const configFile =
```

Comment:
should there be a fallback to tsconfig.json?

Reply:
Nope, I don't set the configFile props if they are not specified. Then the loaders will just use the default

Reply:
cool!